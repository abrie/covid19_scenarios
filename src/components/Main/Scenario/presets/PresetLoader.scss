.preset-loader-button {
  padding: 0.375rem 0.75rem;
  margin: 0 0.5em 0 0;
  font-size: 16px;
  text-transform: uppercase;
}
