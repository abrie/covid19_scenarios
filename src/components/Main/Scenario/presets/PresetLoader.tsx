import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { Button } from 'reactstrap'
import { LoadPresetDialogRecordProps } from './PresetLoaderDialogRow'
import PresetLoaderDialog from './PresetLoaderDialog'
import './PresetLoader.scss'

interface LoadPresetButtonProps {
  data: LoadPresetDialogRecordProps[]
  disabled?: boolean
  onSelect: (id: string) => void
}

const PresetLoader = ({ data, onSelect }: LoadPresetButtonProps) => {
  const { t } = useTranslation()
  const [showDialog, setShowDialog] = useState(false)
  const button = (
    <Button
      className="preset-loader-button"
      color="primary"
      size="sm"
      data-testid="PresetLoaderButton"
      onClick={() => setShowDialog(true)}
    >
      {t('Change scenario')}
    </Button>
  )
  const dialog = (
    <PresetLoaderDialog
      data={data}
      visible={showDialog}
      onLoadButtonClick={(id: string) => {
        onSelect(id)
        setShowDialog(false)
      }}
      onClose={() => {
        setShowDialog(false)
      }}
    />
  )
  return (
    <>
      {button}
      {dialog}
    </>
  )
}

export default PresetLoader
