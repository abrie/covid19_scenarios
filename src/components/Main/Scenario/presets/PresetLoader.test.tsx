import React from 'react'
import { cleanup, fireEvent, render, RenderResult } from '@testing-library/react'
import PresetLoader from './PresetLoader'

const DATA = [
  {
    label: 'First row',
    value: '1',
  },
  {
    label: 'Second row',
    value: '2',
  },
  {
    label: 'Third row',
    value: '3',
  },
]

describe('PresetLoader', () => {
  let onSelect: (id: string) => void
  let wrapper: RenderResult

  beforeEach(() => {
    onSelect = jest.fn()
    wrapper = render(<PresetLoader data={DATA} onSelect={onSelect} />)
  })

  afterEach(cleanup)

  it('matches snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('renders button', () => {
    expect(wrapper.getByTestId('PresetLoaderButton')).not.toBeNull()
    expect(wrapper.getByText('Change scenario')).not.toBeNull()
  })

  it('does not render dialog initially', () => {
    expect(wrapper.queryByTestId('PresetLoaderDialog')).toBeNull()
  })

  it('renders dialog on button click', () => {
    const button = wrapper.getByTestId('PresetLoaderButton')
    fireEvent.click(button)

    expect(wrapper.getByTestId('PresetLoaderDialog')).not.toBeNull()
  })

  it('triggers onSelect on row button click', () => {
    const button = wrapper.getByTestId('PresetLoaderButton')
    fireEvent.click(button)

    const loadButtons = wrapper.getAllByTestId('PresetLoaderDialogRowButton')

    expect(loadButtons).toHaveLength(3)

    fireEvent.click(loadButtons[0]) // click the 1st Load button

    expect(onSelect).toHaveBeenCalledWith('1')
  })
})
