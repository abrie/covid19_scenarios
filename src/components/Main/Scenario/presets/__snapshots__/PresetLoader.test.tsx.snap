// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PresetLoader matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="preset-loader-button btn btn-primary btn-sm"
    data-testid="PresetLoaderButton"
    type="button"
  >
    Change scenario
  </button>
</DocumentFragment>
`;
